c-dependencies/js-compute-runtime/js-compute-builtins.cpp

```diff
@@ -4198,23 +4202,39 @@ bool init_class(JSContext *cx, HandleObject global) {
   return !!GET_atom;
 }
 
+JSObject *create(JSContext *cx, HandleObject requestInstance, RequestHandle request_handle,
+                 BodyHandle body_handle, bool is_downstream);
+
 JSObject *create(JSContext *cx, RequestHandle request_handle, BodyHandle body_handle,
                  bool is_downstream) {
-  RootedObject request(cx, JS_NewObjectWithGivenProto(cx, &class_, proto_obj));
-  if (!request)
+  RootedObject requestInstance(cx, JS_NewObjectWithGivenProto(cx, &class_, proto_obj));
+  if (!requestInstance)
     return nullptr;
 
-  JS::SetReservedSlot(request, Slots::Request, JS::Int32Value(request_handle.handle));
-  JS::SetReservedSlot(request, Slots::Headers, JS::NullValue());
-  JS::SetReservedSlot(request, Slots::Body, JS::Int32Value(body_handle.handle));
-  JS::SetReservedSlot(request, Slots::BodyStream, JS::NullValue());
-  JS::SetReservedSlot(request, Slots::HasBody, JS::FalseValue());
-  JS::SetReservedSlot(request, Slots::BodyUsed, JS::FalseValue());
-  JS::SetReservedSlot(request, Slots::Method, JS::StringValue(GET_atom));
-  JS::SetReservedSlot(request, Slots::CacheOverride, JS::NullValue());
-  JS::SetReservedSlot(request, Slots::IsDownstream, JS::BooleanValue(is_downstream));
+  return Request::create(cx, requestInstance, request_handle, body_handle, is_downstream);
+}
```

Comment:
This method exists for use within the `prepare_downstream_request` function, which should always use the runtime's Request implementation and not one which could be supplied by the applications' JS